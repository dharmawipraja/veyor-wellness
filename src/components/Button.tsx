import { MouseEventHandler, useMemo } from "react";

type ButtonProps = {
  title: string;
  type?: "submit" | "reset" | "button";
  onClick?: MouseEventHandler<HTMLButtonElement>;
  variant?: 'PRIMARY' | 'SECONDARY';
}

const getButtonStyles = (variant: string) => {
  const styleMapping: Record<string, string> = {
    PRIMARY: "text-white bg-black ",
    SECONDARY: "w-32 p-2 text-black bg-white border border-gray-700"
  }

  return styleMapping[variant]
}

const Button = ({ title, type, onClick = () => {}, variant = 'PRIMARY' }: ButtonProps) => {
  const styles = getButtonStyles(variant);

  const renderButton = useMemo(() => (
    <button type={type} className={`${styles} max-w-prose p-2 rounded-md`} onClick={onClick}>
      <span>{title}</span>
    </button>
  ), [])

  return (
    renderButton
  )
}

export default Button