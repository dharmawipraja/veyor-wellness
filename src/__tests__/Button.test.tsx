import { render, screen, fireEvent } from '@testing-library/react';
import Button from '../components/Button';

describe('Button component', () => {
  it("should match snapshot", () => {
    const handleClick = jest.fn();
    const { asFragment } = render(<Button title="Click Me" onClick={handleClick} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should renders with correct title', () => {
    render(<Button title="Click Me" />);

    const buttonElement = screen.getByText(/Click Me/i);
    expect(buttonElement).toBeInTheDocument();
  });

  it('should has the correct default type', () => {
    render(<Button title="Click Me" />);
    
    const buttonElement = screen.getByText(/Click Me/i);
    expect(buttonElement).toHaveAttribute('type', 'button');
  });

  it('should calls onClick handler when clicked', () => {
    const handleClick = jest.fn();
    render(<Button title="Click Me" onClick={handleClick} />);

    const buttonElement = screen.getByText(/Click Me/i);
    fireEvent.click(buttonElement);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('should applies correct styles for PRIMARY variant', () => {
    render(<Button title="Click Me" variant="PRIMARY" />);
    
    const buttonElement = screen.getByText(/Click Me/i);
    expect(buttonElement).toHaveClass('text-white bg-black');
  });

  it('should applies correct styles for SECONDARY variant', () => {
    render(<Button title="Click Me" variant="SECONDARY" />);
    
    const buttonElement = screen.getByText(/Click Me/i);
    expect(buttonElement).toHaveClass('text-black bg-white border border-gray-700');
  });

  it('should renders with custom type', () => {
    render(<Button title="Submit" type="submit" />);
    
    const buttonElement = screen.getByText(/Submit/i);
    expect(buttonElement).toHaveAttribute('type', 'submit');
  });
});
