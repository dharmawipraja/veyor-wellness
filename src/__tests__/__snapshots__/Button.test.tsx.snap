// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center min-w-32 justify-center py-2 px-5 rounded-md text-white bg-black"
    type="button"
  >
    Click Me
  </button>
</DocumentFragment>
`;
